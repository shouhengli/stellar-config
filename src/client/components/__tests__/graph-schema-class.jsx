const React = require('react');
const renderer = require('react-test-renderer');
const { mount, shallow } = require('enzyme');
const { fromJS } = require('immutable');
const Class = require('../graph-schema-class.jsx');

describe('component graph-schema-class', () => {
  let props;

  beforeEach(() => {
    props = {
      ClassName: 'div',
      ClassArc: 'div',
      ClassPropName: 'div',
      ClassPropTooltip: 'div',
      cls: fromJS({
        name: 'Person',
        globalIndex: '1',
        props: { 10: { name: 'name' } },
        x: 0,
        y: 0,
        tooltipVisibleProp: null,
        outerRadius: 75
      }),
      handleMouseEnter: jest.fn(),
      handleMouseLeave: jest.fn(),
      handleComponentWillUnmount: jest.fn()
    };
  });

  test('can be rendered', () => {
    const tree = renderer.create(<Class {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('can handle componentWillUnmount event', () => {
    const wrapper = mount(<Class {...props} />);
    wrapper.unmount();

    expect(props.handleComponentWillUnmount).toHaveBeenCalledTimes(1);
    expect(props.handleComponentWillUnmount).toHaveBeenCalledWith(props.cls);
  });

  test('can handle mouse-enter event', () => {
    const wrapper = shallow(<Class {...props} />);
    wrapper.simulate('mouseenter');

    expect(props.handleMouseEnter).toHaveBeenCalledTimes(1);
    expect(props.handleMouseEnter).toHaveBeenCalledWith(props.cls);
  });

  test('can handle mouse-leave event', () => {
    const wrapper = shallow(<Class {...props} />);
    wrapper.simulate('mouseleave');

    expect(props.handleMouseLeave).toHaveBeenCalledTimes(1);
    expect(props.handleMouseLeave).toHaveBeenCalledWith(props.cls);
  });
});
