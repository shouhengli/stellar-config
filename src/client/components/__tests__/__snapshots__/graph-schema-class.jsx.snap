// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component graph-schema-class can be rendered 1`] = `
<g
  className="graph-schema-class"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  transform="translate(0, 0)"
>
  <div
    fontSize={16}
    name="Person"
    radius={75}
  />
  <g
    transform="rotate(180)"
  >
    <div
      classPropName="name"
      globalIndex="1"
      path="M4.592425496802574e-15,-75A75,75,0,1,1,-4.592425496802574e-15,75A75,75,0,1,1,4.592425496802574e-15,-75M-1.3777276490407723e-14,-75A75,75,0,1,0,1.3777276490407723e-14,75A75,75,0,1,0,-1.3777276490407723e-14,-75Z"
    />
    <div
      classPropName="name"
      classPropNameArcPath="M4.592425496802574e-15,-75A75,75,0,1,1,-4.592425496802574e-15,75A75,75,0,1,1,4.592425496802574e-15,-75Z"
      classPropNameRadius={75}
      classPropNameVisibility="hidden"
      clipPath="M4.592425496802574e-15,-75A75,75,0,1,1,-4.592425496802574e-15,75A75,75,0,1,1,4.592425496802574e-15,-75M-1.3777276490407723e-14,-75A75,75,0,1,0,1.3777276490407723e-14,75A75,75,0,1,0,-1.3777276490407723e-14,-75Z"
      fontSize={16}
      globalIndex="1"
      id={undefined}
      rotation={0}
    />
    <div
      classPropName="name"
      classPropTooltipArcPath="M5.204748896376251e-15,-85A85,85,0,1,1,-5.204748896376251e-15,85A85,85,0,1,1,5.204748896376251e-15,-85Z"
      classPropTooltipRadius={85}
      id={undefined}
      rotation={0}
      visible={false}
    />
  </g>
</g>
`;
